03-write-wdl.Rmd

```diff
@@ -0,0 +1,167 @@
+ottrpal::include_slide("https://docs.google.com/presentation/d/1o2XnuMbqWVLf4XrsXolIQ7ulfnMlpJlrUxN0Y8aLIVQ/edit#slide=id.g139bf26eaed_0_27")
+```
+
+Click on the Find a Workflow card.
+
+```{r, echo=FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1o2XnuMbqWVLf4XrsXolIQ7ulfnMlpJlrUxN0Y8aLIVQ/edit#slide=id.g139bf26eaed_0_1")
+```
+
+Select the Broad Methods Repository option.
+
+```{r, echo=FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1o2XnuMbqWVLf4XrsXolIQ7ulfnMlpJlrUxN0Y8aLIVQ/edit#slide=id.g139bf26eaed_0_6")
+```
+
+Click Create New Method.
+
+```{r, echo=FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1o2XnuMbqWVLf4XrsXolIQ7ulfnMlpJlrUxN0Y8aLIVQ/edit#slide=id.g139bf26eaed_0_11")
+```
+
+Add a namespace to the first text box to organize your WDLs.  
+Your username (prepended with your lab name) is a reasonable namespace as this must be unique across all of Broad Methods Repository.  
+Afterwards, add a name such as `wdl101` to name your WDL.
+
+```{r, echo=FALSE}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1o2XnuMbqWVLf4XrsXolIQ7ulfnMlpJlrUxN0Y8aLIVQ/edit#slide=id.g139bf26eaed_0_16")
+```
+
+## Write WDL101 Training Example
+
+Let's now create a basic WDL!  This simple "Hello, World!" style workflow will take as input a string, call a single task, and save the output of that task to your workspace bucket.  The task that is called will run the Bash `echo` command to print the input string to `stdout`.
+
+First note that we are using the [WDL 1.0 spec](https://github.com/openwdl/wdl/tree/main/versions).
+
+```
+version 1.0
+```
+
+Let's add a workflow `HelloInput` that calls a single task `WriteGreeting`.
+
+```
+version 1.0
+workflow HelloInput {
+}
+
+task WriteGreeting{
+}
+```
+
+Now let's define an input string for the workflow and store it in a variable named `name_input`.  
+That information will be used when calling the task `WriteGreeting` by passing `name_input` to `name_for_greeting`.  
+Lastly, the workflow will store what is returned by the task in a File labeled `final_output`.
+
+```
+version 1.0
+workflow HelloInput {
+  input {
+    String name_input
+  }
+  call WriteGreeting {
+    input: 
+      name_for_greeting = name_input
+  }
+  output {
+    File final_output = WriteGreeting.Greeting_output
+  }
+}
+
+task WriteGreeting {
+}
+```
+
+Putting it all together, we now define `name_for_greeting`, the `command`,  `Greeting_output`, and `docker` for the task.
+
+```
+version 1.0
+workflow HelloInput {
+  input {
+    String name_input
+  }
+  call WriteGreeting {
+    input: 
+      name_for_greeting = name_input
+  }
+  output {
+    File final_output = WriteGreeting.Greeting_output
+  }
+}
+
+task WriteGreeting {
+  input {
+    String name_for_greeting
+  }
+  command <<<
+    echo 'hello ~{name_for_greeting}!'
+  >>>
+  output {
+    File Greeting_output = stdout()
+  }
+  runtime {
+    docker: 'ubuntu:latest'
+  }
+}
```

Comment:
Liz had us do the `WriteGreeting` part first. Any particular reason you've arranged it this way? I think it's a bit harder to explain the workflow task when you don't know what the task actually is.